File: fly-promo-publisher-flyWU/src/main/java/io/github/riojano0/flypromoflyWU/model/CountryProvider.java
package io.github.riojano0.flypromoflyWU.model;

import io.github.riojano0.flypromokafka.core.model.Country;

public enum CountryProvider implements Country {

   ARGENTINA(() -> "ARG"),
   COLOMBIA(() -> "COL"),
   BRAZIL(() -> "BAR"),
   MEXICO(() -> "MEX"),
   CHILE (() -> "CHL");

   private final Country country;

   CountryProvider(Country country) {
      this.country = country;
   }

   @Override
   public String getISO3() {
      return country.getISO3();
   }
}

File: fly-promo-publisher-fly2EU/src/main/kotlin/io/github/riojano0/flypromofly2EU/model/CountryProvider.kt
package io.github.riojano0.flypromofly2EU.model

import io.github.riojano0.flypromokafka.core.model.Country

enum class CountryProvider(private val country: Country) : Country {

    SPAIN(Country { "SPA" }),
    PORTUGAL(Country { "POR" }),
    ITALY(Country { "ITA" }),
    GERMANY(Country { "GER" }),
    FRANCE(Country { "FRA" });

    override fun getISO3(): String {
        return this.country.isO3
    }
}